grace/run.py

```diff
@@ -0,0 +1,263 @@
+            keep_edge_unknown_labels=config.keep_edge_unknown_labels,
+            transform=transform_method,
+        )
+
+        # Process the (sub)graph data into torch_geometric dataset:
+        target_list, subgraph_dataset = [], []
+        desc = "Extracting patch features from images... "
+        for _, target in tqdm(input_data, desc=desc, disable=not verbose):
+            file_name = target["metadata"]["image_filename"]
+            LOGGER.info(f"Processing file: {file_name}")
+
+            # Store the valid graph list:
+            target_list.append(target)
+
+            # Chop graph into subgraphs & store:
+            graphs = dataset_from_graph(target["graph"], mode=graph_processing)
+            subgraph_dataset.extend(graphs)
+
+        return target_list, subgraph_dataset
+
+    # Create a transform function with frozen 'in_train_mode' parameter:
+    transform_train_mode = partial(transform, in_train_mode=True)
+    transform_valid_mode = partial(transform, in_train_mode=False)
+
+    # Read the respective datasets:
+    _, train_dataset = prepare_dataset(
+        config.train_image_dir,
+        config.train_grace_dir,
+        transform_train_mode,
+    )
+    valid_target_list, valid_dataset = prepare_dataset(
+        config.valid_image_dir,
+        config.valid_grace_dir,
+        transform_valid_mode,
+    )
+    infer_target_list, _ = prepare_dataset(
+        config.infer_image_dir,
+        config.infer_grace_dir,
+        transform_valid_mode,
+    )
+
+    # Define the GNN classifier model:
+    classifier = GNNClassifier().get_model(
+        config.classifier_type,
+        input_channels=config.feature_dim,
+        hidden_channels=config.hidden_channels,
+        dropout=config.dropout,
+        node_output_classes=config.num_node_classes,
+        edge_output_classes=config.num_edge_classes,
+    )
+
+    # Perform the training:
+    train_model(
+        classifier,
+        train_dataset,
+        valid_dataset,
+        valid_target_list,
+        epochs=config.epochs,
+        batch_size=config.batch_size,
+        learning_rate=config.learning_rate,
+        log_dir=run_dir,
+        metrics=config.metrics_classifier,
+        tensorboard_update_frequency=config.tensorboard_update_frequency,
+        valid_graph_ploter_frequency=config.valid_graph_ploter_frequency,
+    )
+
+    # Save the trained model:
+    model_save_fn = run_dir / "classifier.pt"
+    torch.save(classifier, model_save_fn)
+    write_config_file(config, filetype="json")
+    write_config_file(config, filetype="yaml")
+
+    # Project the TSNE manifold:
+    if config.visualise_tsne_manifold is True:
+        # TODO: Implement
+        pass
+
+    # Animate the validation outputs:
+    if config.animate_valid_progress is True:
+        animate_entire_valid_set(run_dir / "valid", verbose=False)
+
+    # Run inference on the final, trained model on unseen data:
+    GLP = GraphLabelPredictor(model_save_fn)
```

Comment:
Thinking about what @quantumjot mentioned today about separating training from inference, this is exactly where this could happen. 

To maintain the flexibility of the run script in the short-term we could say if a `model_save_fn` is given in the config file (and maybe an inference-only flag), then run script from here. Further in the line (when we know Grace is working as expected) we can think how es best to abstract each piece.

This comment is not for this PR, just to keep this in mind for the next step.